src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/nonassociations/NonAssociationService.kt

```diff
@@ -348,7 +348,13 @@ class NonAssociationService(
           effectiveDate = detail.effectiveDate,
           expiryDate = detail.expiryDate,
           authorisedBy = detail.authorisedBy,
+          updatedBy = detail.updatedBy,
           comment = detail.comment,
         )
       }
 }
+
+private val OffenderNonAssociationDetail.updatedBy: String
+  get() {
+    return modifiedBy ?: createdBy!!
```

Comment:
annoyingly have to add !! as createdBy is set to nullable

Reply:
Just checked visits and there the field is not nullable, but initialised to be blank - maybe try that?